src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -341,6 +341,30 @@ internal static unsafe bool EndsWithOrdinalHelper(ReadOnlySpan<char> span, ReadO
             return OrdinalHelper(span.Slice(span.Length - value.Length), value, value.Length);
         }
 
+        /// <summary>
+        /// Helper method for MemoryExtensions.AsSpan(T[] array, int start).
+        /// </summary>
+        public static Span<T> AsSpan<T>(T[] array, int start)
+        {
+            if (array == null)
+            {
+                if (start != 0)
+                    ThrowHelper.ThrowArgumentOutOfRangeException();
+                return default;
+            }
+            if (default(T) == null && array.GetType() != typeof(T[]))
+                ThrowHelper.ThrowArrayTypeMismatchException();
+            if ((uint)start > (uint)array.Length)
+                ThrowHelper.ThrowArgumentOutOfRangeException();
+
+            return new Span<T>(ref Unsafe.Add(ref Unsafe.As<byte, T>(ref array.GetRawSzArrayData()), start), array.Length - start);
+        }
+
+        /// <summary>
+        /// Helper method for MemoryExtensions.AsMemory(T[] array, int start).
+        /// </summary>
+        public static Memory<T> AsMemory<T>(T[] array, int start) => new Memory<T>(array, start);
```

Comment:
Is this still needed or can MemoryExtensions call the internal ctor directly?

Reply:
I would be surprised if the JIT couldn't inline all the indirection away here so unless that's incorrect, I'd rather stay with the flow where "private apis" to CoreFx are exposed in nongeneric Span and the api type Memory's surface area matches the reference api surface area. Less confusing that way.